src/benchmark_ocl.cpp

```diff
@@ -144,125 +119,106 @@ int main(void)
+	ocl_helper.write_device_info(std::cerr);
+
+	// allocate OpenCL device memory // DONE: replace with create_buffer() from noma::ocl, use C++ interface
+	noma::ocl::buffer hamiltonian_ocl = ocl_helper.create_buffer(CL_MEM_READ_ONLY,  size_hamiltonian_byte);
+	noma::ocl::buffer sigma_in_ocl    = ocl_helper.create_buffer(CL_MEM_READ_WRITE, size_sigma_byte      );
+	noma::ocl::buffer sigma_out_ocl   = ocl_helper.create_buffer(CL_MEM_READ_WRITE, size_sigma_byte      );
 
 	// function to build and set-up a kernel
 	auto prepare_kernel = [&](const std::string& file_name, const std::string& kernel_name, const std::string& compile_options)
 	{
 		// build kernel
 		time::rep build_time = 0.0;
-		cl_program prog;
+		cl::Program prog;
 		{
 			time::timer t;
-			prog = cluBuildSourceFromFile(file_name.c_str(), compile_options.c_str(), &err);
+			prog = ocl_helper.create_program_from_file(file_name, "", compile_options);
 			build_time = t.elapsed();
 		}
 		
 		std::stringstream time_ss;
 		time_ss << std::scientific << build_time;
 		std::cout << "build_time\t" << kernel_name << "\t" << time_ss.str() << std::endl;
 		
-		if (ocl_error_handler(err, "cluBuildSourceFromFile()", false))
-		{
-			std::cerr << "OpenCL build log for: " << file_name << std::endl
-			          << cluGetBuildErrors(prog) << std::endl
-			          << "--- end of build log ---" << std::endl;
-			exit(-1);
-		}
-		cl_kernel kernel = clCreateKernel(prog, kernel_name.c_str(), &err);
-		ocl_error_handler(err, "clCreateKernel()");
+		// get kernel from programm using C++ OCL API
+		cl::Kernel kernel(prog, kernel_name.c_str(), &err);
+		noma::ocl::error_handler(err, "Error creating kernel.");
 
 		// set kernel arguments
-		err = clSetKernelArg(kernel, 0, sizeof(cl_mem), static_cast<const void*>(&sigma_in_ocl));
-		ocl_error_handler(err, "clSetKernelArg(0)");
-		err = clSetKernelArg(kernel, 1, sizeof(cl_mem), static_cast<const void*>(&sigma_out_ocl));
-		ocl_error_handler(err, "clSetKernelArg(1)");
-		err = clSetKernelArg(kernel, 2, sizeof(cl_mem), static_cast<const void*>(&hamiltonian_ocl));
-		ocl_error_handler(err, "clSetKernelArg(2)");
+		err = kernel.setArg(0, static_cast<cl::Buffer>(sigma_in_ocl));
+		noma::ocl::error_handler(err, "kernel.setArg(0)");
+		err = kernel.setArg(1, static_cast<cl::Buffer>(sigma_out_ocl));
+		noma::ocl::error_handler(err, "kernel.setArg(1)");
+		err = kernel.setArg(2, static_cast<cl::Buffer>(hamiltonian_ocl));
+		noma::ocl::error_handler(err, "kernel.setArg(2)");
 		// GCC bug work-around (#ifdef __GNUC__) & type conversion
 		// http://stackoverflow.com/questions/19616610/c11-lambda-doesnt-take-const-variable-by-reference-why
+		// TODO: Is it okay to remove const keyword? Reference:
+		// Internet:                                    cl_int setArg(cl_uint index, size_type size, const void *argPtr)
+		// thirdparty/ocl/thirdparty/include/CL/cl.hpp: cl_int setArg(cl_uint index,  ::size_t size,       void* argPtr)
```

Comment:
NOTE: We keep the tmp variables of defined-length-types here, as we have to make sure, the byte size still matches the OpenCL kernel signature types.